### ISSUE TITLE
kernel BUG at fs/bcachefs/btree_iter.c:1332

### ISSUE THREAD

**Issue by marcin-github** [koverstreet/bcachefs#296]
I'm not sure if it's known bug, if is known then please close it.
commit 60fbf06f49679fdb

```
[ 3491.296582] ------------[ cut here ]------------
[ 3491.296698] kernel BUG at fs/bcachefs/btree_iter.c:1332!
[ 3491.296727] invalid opcode: 0000 [#1] SMP DEBUG_PAGEALLOC NOPTI
[ 3491.296734] CPU: 1 PID: 140118 Comm: burp Not tainted 5.13.0-00858-g60fbf06f4967 #14
[ 3491.296741] Hardware name: MSI MS-7982/B150M PRO-VDH (MS-7982), BIOS 3.H0 07/10/2018
[ 3491.296746] RIP: 0010:__btree_iter_traverse_all+0x28e/0x390
[ 3491.296755] Code: 00 00 00 0f 85 17 01 00 00 48 83 c4 78 44 89 f0 5b 5d 41 5c 41 5d 41 5e 41 5f c3 be 01 00 00 00 e8 67 fc ff ff e9 56 fe ff ff <0f> 0b 48 8b 73 08 65 8b 05 2d 92 bb 5c 89 c0 48 0f a3 05 ab 3d 0e
[ 3491.296765] RSP: 0018:ffffba80507e7838 EFLAGS: 00010202
[ 3491.296771] RAX: ffff939c650a4000 RBX: ffffba80507e7a38 RCX: 0000000000000001
[ 3491.296776] RDX: 0000000000000002 RSI: 0000000000000005 RDI: ffff939c650a4000
[ 3491.296782] RBP: ffffba80507e7848 R08: 0000000000000000 R09: 0000000000000000
[ 3491.296787] R10: 0000000000000001 R11: 0000000000000000 R12: 0000000000000001
[ 3491.296792] R13: ffffffffffffffff R14: 0000000000000000 R15: ffff939c650a41d0
[ 3491.296797] FS:  00007f4404ed3b80(0000) GS:ffff939ce4a00000(0000) knlGS:0000000000000000
[ 3491.296804] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 3491.296808] CR2: 00000000d6f1b000 CR3: 00000001a9be0001 CR4: 00000000003706e0
[ 3491.296814] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 3491.296819] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[ 3491.296824] Call Trace:
[ 3491.296827]  ? __bch2_create+0x136/0x500
[ 3491.296834]  ? bch2_trans_unlink_iters+0xc6/0x200
[ 3491.296840]  bch2_trans_begin+0xfe/0x120
[ 3491.296847]  __bch2_create+0x136/0x500
[ 3491.296855]  ? __bch2_create+0x10f/0x500
[ 3491.296862]  bch2_create+0x1c/0x40
[ 3491.296867]  lookup_open.isra.0+0x303/0x640
[ 3491.296875]  ? __wake_up_common_lock+0x62/0xc0
[ 3491.296882]  path_openat+0x240/0x900
[ 3491.296888]  do_filp_open+0x99/0x140
[ 3491.296894]  ? lock_release+0x26c/0x3c0
[ 3491.296901]  ? _raw_spin_unlock+0x1f/0x30
[ 3491.296907]  ? alloc_fd+0x12c/0x1f0
[ 3491.296914]  do_sys_openat2+0x97/0x150
[ 3491.296921]  __x64_sys_openat+0x54/0x90
[ 3491.296927]  do_syscall_64+0x5e/0xb0
[ 3491.296932]  ? filename_create+0xc3/0x160
[ 3491.296939]  ? rcu_read_lock_sched_held+0x3f/0x70
[ 3491.296945]  ? lockdep_hardirqs_on_prepare+0xd4/0x170
[ 3491.296950]  ? do_syscall_64+0x6e/0xb0
[ 3491.296955]  entry_SYSCALL_64_after_hwframe+0x44/0xae
[ 3491.296961] RIP: 0033:0x7f4404fe6077
[ 3491.296966] Code: 25 00 00 41 00 3d 00 00 41 00 74 47 64 8b 04 25 18 00 00 00 85 c0 75 6b 44 89 e2 48 89 ee bf 9c ff ff ff b8 01 01 00 00 0f 05 <48> 3d 00 f0 ff ff 0f 87 95 00 00 00 48 8b 4c 24 28 64 48 2b 0c 25
[ 3491.296976] RSP: 002b:00007ffcca417a40 EFLAGS: 00000246 ORIG_RAX: 0000000000000101
[ 3491.297039] RAX: ffffffffffffffda RBX: 0000560da9bf84c0 RCX: 00007f4404fe6077
[ 3491.297044] RDX: 0000000000020041 RSI: 0000560da9bf84e0 RDI: 00000000ffffff9c
[ 3491.297049] RBP: 0000560da9bf84e0 R08: 0000000000000007 R09: 0000560da9bf8480
[ 3491.297054] R10: 00000000000001b6 R11: 0000000000000246 R12: 0000000000020041
[ 3491.297059] R13: 0000560da9bc3b80 R14: 0000000000000000 R15: 00007ffcca417cd0
[ 3491.297120] Modules linked in: xt_nat xt_tcpudp veth netconsole configfs zram xt_conntrack xt_MASQUERADE nf_conntrack_netlink nfnetlink xfrm_user xfrm_algo xt_addrtype iptable_filter iptable_nat ip_tables x_tables bpfil
ter br_netfilter tun nfsd auth_rpcgss nfs_acl lockd grace overlay sunrpc af_packet bridge stp llc pcrypt nf_nat_ftp nf_conntrack_ftp nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 coretemp hwmon i915 x86_pkg_temp_therma
l intel_powerclamp kvm_intel kvm i2c_algo_bit intel_gtt iosf_mbi drm_kms_helper i2c_i801 i2c_smbus cfbfillrect snd_pcsp irqbypass syscopyarea rapl cfbimgblt intel_cstate sysfillrect sysimgblt snd_pcm fb_sys_fops cfbcopyare
a fb font fbdev snd_timer intel_uncore drm snd r8169 soundcore realtek mdio_devres of_mdio fixed_phy intel_lpss_pci libphy intel_lpss mei_me bfq i2c_core idma64 mei drm_panel_orientation_quirks intel_pch_thermal virt_dma m
fd_core thermal fan video backlight acpi_pad button raid1 md_mod crct10dif_pclmul crc32_pclmul crc32c_intel
[ 3491.297176]  ghash_clmulni_intel mpt3sas aesni_intel crypto_simd xhci_pci raid_class scsi_transport_sas cryptd xhci_hcd dm_mod
[ 3491.297280] ---[ end trace 449aff45c1ad5d1b ]---
[ 3491.297285] RIP: 0010:__btree_iter_traverse_all+0x28e/0x390
[ 3491.297291] Code: 00 00 00 0f 85 17 01 00 00 48 83 c4 78 44 89 f0 5b 5d 41 5c 41 5d 41 5e 41 5f c3 be 01 00 00 00 e8 67 fc ff ff e9 56 fe ff ff <0f> 0b 48 8b 73 08 65 8b 05 2d 92 bb 5c 89 c0 48 0f a3 05 ab 3d 0e
[ 3491.297301] RSP: 0018:ffffba80507e7838 EFLAGS: 00010202
[ 3491.297306] RAX: ffff939c650a4000 RBX: ffffba80507e7a38 RCX: 0000000000000001
[ 3491.297311] RDX: 0000000000000002 RSI: 0000000000000005 RDI: ffff939c650a4000
[ 3491.297317] RBP: ffffba80507e7848 R08: 0000000000000000 R09: 0000000000000000
[ 3491.297322] R10: 0000000000000001 R11: 0000000000000000 R12: 0000000000000001
[ 3491.297327] R13: ffffffffffffffff R14: 0000000000000000 R15: ffff939c650a41d0
[ 3491.297333] FS:  00007f4404ed3b80(0000) GS:ffff939ce4a00000(0000) knlGS:0000000000000000
[ 3491.297340] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 3491.297347] CR2: 00000000d6f1b000 CR3: 00000001a9be0001 CR4: 00000000003706e0
[ 3491.297353] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 3491.297358] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
```

```
# scripts/faddr2line vmlinux __btree_iter_traverse_all+0x28e/0x390
__btree_iter_traverse_all+0x28e/0x390:
__btree_iter_traverse_all at fs/bcachefs/btree_iter.c:1332 (discriminator 1)
```

**Comment by marcin-github** [comment]
Is decode_stracktrace.sh useful?
```
[ 3491.296582] ------------[ cut here ]------------
[ 3491.296698] kernel BUG at fs/bcachefs/btree_iter.c:1332!
[ 3491.296727] invalid opcode: 0000 [#1] SMP DEBUG_PAGEALLOC NOPTI
[ 3491.296734] CPU: 1 PID: 140118 Comm: burp Not tainted 5.13.0-00858-g60fbf06f4967 #14
[ 3491.296741] Hardware name: MSI MS-7982/B150M PRO-VDH (MS-7982), BIOS 3.H0 07/10/2018
[ 3491.296746] RIP: 0010:__btree_iter_traverse_all (fs/bcachefs/btree_iter.c:1332 (discriminator 1))
[ 3491.296755] Code: 00 00 00 0f 85 17 01 00 00 48 83 c4 78 44 89 f0 5b 5d 41 5c 41 5d 41 5e 41 5f c3 be 01 00 00 00 e8 67 fc ff ff e9 56 fe ff ff <0f> 0b 48 8b 73 08 65 8b 05 2d 92 bb 5c 89 c0 48 0f a3 05 ab 3d 0e
All code
========
   0:   00 00                   add    %al,(%rax)
   2:   00 0f                   add    %cl,(%rdi)
   4:   85 17                   test   %edx,(%rdi)
   6:   01 00                   add    %eax,(%rax)
   8:   00 48 83                add    %cl,-0x7d(%rax)
   b:   c4                      (bad)
   c:   78 44                   js     0x52
   e:   89 f0                   mov    %esi,%eax
  10:   5b                      pop    %rbx
  11:   5d                      pop    %rbp
  12:   41 5c                   pop    %r12
  14:   41 5d                   pop    %r13
  16:   41 5e                   pop    %r14
  18:   41 5f                   pop    %r15
  1a:   c3                      retq
  1b:   be 01 00 00 00          mov    $0x1,%esi
  20:   e8 67 fc ff ff          callq  0xfffffffffffffc8c
  25:   e9 56 fe ff ff          jmpq   0xfffffffffffffe80
  2a:*  0f 0b                   ud2             <-- trapping instruction
  2c:   48 8b 73 08             mov    0x8(%rbx),%rsi
  30:   65 8b 05 2d 92 bb 5c    mov    %gs:0x5cbb922d(%rip),%eax        # 0x5cbb9264
  37:   89 c0                   mov    %eax,%eax
  39:   48                      rex.W
  3a:   0f                      .byte 0xf
  3b:   a3                      .byte 0xa3
  3c:   05                      .byte 0x5
  3d:   ab                      stos   %eax,%es:(%rdi)
  3e:   3d                      .byte 0x3d
  3f:   0e                      (bad)

Code starting with the faulting instruction
===========================================
   0:   0f 0b                   ud2
   2:   48 8b 73 08             mov    0x8(%rbx),%rsi
   6:   65 8b 05 2d 92 bb 5c    mov    %gs:0x5cbb922d(%rip),%eax        # 0x5cbb923a
   d:   89 c0                   mov    %eax,%eax
   f:   48                      rex.W
  10:   0f                      .byte 0xf
  11:   a3                      .byte 0xa3
  12:   05                      .byte 0x5
  13:   ab                      stos   %eax,%es:(%rdi)
  14:   3d                      .byte 0x3d
  15:   0e                      (bad)
[ 3491.296765] RSP: 0018:ffffba80507e7838 EFLAGS: 00010202
[ 3491.296771] RAX: ffff939c650a4000 RBX: ffffba80507e7a38 RCX: 0000000000000001
[ 3491.296776] RDX: 0000000000000002 RSI: 0000000000000005 RDI: ffff939c650a4000
[ 3491.296782] RBP: ffffba80507e7848 R08: 0000000000000000 R09: 0000000000000000
[ 3491.296787] R10: 0000000000000001 R11: 0000000000000000 R12: 0000000000000001
[ 3491.296792] R13: ffffffffffffffff R14: 0000000000000000 R15: ffff939c650a41d0
[ 3491.296797] FS:  00007f4404ed3b80(0000) GS:ffff939ce4a00000(0000) knlGS:0000000000000000
[ 3491.296804] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 3491.296808] CR2: 00000000d6f1b000 CR3: 00000001a9be0001 CR4: 00000000003706e0
[ 3491.296814] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 3491.296819] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[ 3491.296824] Call Trace:
[ 3491.296827] ? __bch2_create (fs/bcachefs/fs.c:285)
[ 3491.296834] ? bch2_trans_unlink_iters (fs/bcachefs/btree_iter.c:2142 fs/bcachefs/btree_iter.c:2201 fs/bcachefs/btree_iter.c:2513)
[ 3491.296840] bch2_trans_begin (fs/bcachefs/btree_iter.c:2560)
[ 3491.296847] __bch2_create (fs/bcachefs/fs.c:285)
[ 3491.296855] ? __bch2_create (fs/bcachefs/fs.c:280 (discriminator 4))
[ 3491.296862] bch2_create (fs/bcachefs/fs.c:391 fs/bcachefs/fs.c:402)
[ 3491.296867] lookup_open.isra.0 (fs/namei.c:3215)
[ 3491.296875] ? __wake_up_common_lock (kernel/sched/wait.c:138 (discriminator 1))
[ 3491.296882] path_openat (fs/namei.c:3286 fs/namei.c:3491)
[ 3491.296888] do_filp_open (fs/namei.c:3522)
[ 3491.296894] ? lock_release (kernel/locking/lockdep.c:321 kernel/locking/lockdep.c:5198 kernel/locking/lockdep.c:5534)
[ 3491.296901] ? _raw_spin_unlock (./arch/x86/include/asm/preempt.h:85 ./include/linux/spinlock_api_smp.h:152 kernel/locking/spinlock.c:183)
[ 3491.296907] ? alloc_fd (fs/file.c:525 (discriminator 13))
[ 3491.296914] do_sys_openat2 (fs/open.c:1188)
[ 3491.296921] __x64_sys_openat (fs/open.c:1214)
[ 3491.296927] do_syscall_64 (arch/x86/entry/common.c:47)
[ 3491.296932] ? filename_create (fs/namei.c:3628)
[ 3491.296939] ? rcu_read_lock_sched_held (kernel/rcu/update.c:125)
[ 3491.296945] ? lockdep_hardirqs_on_prepare (kernel/locking/lockdep.c:4109 kernel/locking/lockdep.c:4169)
[ 3491.296950] ? do_syscall_64 (arch/x86/entry/common.c:58)
[ 3491.296955] entry_SYSCALL_64_after_hwframe (arch/x86/entry/entry_64.S:112)
[ 3491.296961] RIP: 0033:0x7f4404fe6077
[ 3491.296966] Code: 25 00 00 41 00 3d 00 00 41 00 74 47 64 8b 04 25 18 00 00 00 85 c0 75 6b 44 89 e2 48 89 ee bf 9c ff ff ff b8 01 01 00 00 0f 05 <48> 3d 00 f0 ff ff 0f 87 95 00 00 00 48 8b 4c 24 28 64 48 2b 0c 25
All code
========
   0:   25 00 00 41 00          and    $0x410000,%eax
   5:   3d 00 00 41 00          cmp    $0x410000,%eax
   a:   74 47                   je     0x53
   c:   64 8b 04 25 18 00 00    mov    %fs:0x18,%eax
  13:   00
  14:   85 c0                   test   %eax,%eax
  16:   75 6b                   jne    0x83
  18:   44 89 e2                mov    %r12d,%edx
  1b:   48 89 ee                mov    %rbp,%rsi
  1e:   bf 9c ff ff ff          mov    $0xffffff9c,%edi
  23:   b8 01 01 00 00          mov    $0x101,%eax
  28:   0f 05                   syscall
  2a:*  48 3d 00 f0 ff ff       cmp    $0xfffffffffffff000,%rax         <-- trapping instruction
  30:   0f 87 95 00 00 00       ja     0xcb
  36:   48 8b 4c 24 28          mov    0x28(%rsp),%rcx
  3b:   64                      fs
  3c:   48                      rex.W
  3d:   2b                      .byte 0x2b
  3e:   0c 25                   or     $0x25,%al

Code starting with the faulting instruction
===========================================
   0:   48 3d 00 f0 ff ff       cmp    $0xfffffffffffff000,%rax
   6:   0f 87 95 00 00 00       ja     0xa1
   c:   48 8b 4c 24 28          mov    0x28(%rsp),%rcx
  11:   64                      fs
  12:   48                      rex.W
  13:   2b                      .byte 0x2b
  14:   0c 25                   or     $0x25,%al
[ 3491.296976] RSP: 002b:00007ffcca417a40 EFLAGS: 00000246 ORIG_RAX: 0000000000000101
[ 3491.297039] RAX: ffffffffffffffda RBX: 0000560da9bf84c0 RCX: 00007f4404fe6077
[ 3491.297044] RDX: 0000000000020041 RSI: 0000560da9bf84e0 RDI: 00000000ffffff9c
[ 3491.297049] RBP: 0000560da9bf84e0 R08: 0000000000000007 R09: 0000560da9bf8480
[ 3491.297054] R10: 00000000000001b6 R11: 0000000000000246 R12: 0000000000020041
[ 3491.297059] R13: 0000560da9bc3b80 R14: 0000000000000000 R15: 00007ffcca417cd0
[ 3491.297120] Modules linked in: xt_nat xt_tcpudp veth netconsole configfs zram xt_conntrack xt_MASQUERADE nf_conntrack_netlink nfnetlink xfrm_user xfrm_algo xt_addrtype iptable_filter iptable_nat ip_tables x_tables bpfilter br_netfilter tun nfsd auth_rpcgss nfs_acl lockd grace overlay sunrpc af_packet bridge stp llc pcrypt nf_nat_ftp nf_conntrack_ftp nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 coretemp hwmon i915 x86_pkg_temp_thermal intel_powerclamp kvm_intel kvm i2c_algo_bit intel_gtt iosf_mbi drm_kms_helper i2c_i801 i2c_smbus cfbfillrect snd_pcsp irqbypass syscopyarea rapl cfbimgblt intel_cstate sysfillrect sysimgblt snd_pcm fb_sys_fops cfbcopyarea fb font fbdev snd_timer intel_uncore drm snd r8169 soundcore realtek mdio_devres of_mdio fixed_phy intel_lpss_pci libphy intel_lpss mei_me bfq i2c_core idma64 mei drm_panel_orientation_quirks intel_pch_thermal virt_dma mfd_core thermal fan video backlight acpi_pad button raid1 md_mod crct10dif_pclmul crc32_pclmul crc32c_intel
[ 3491.297176]  ghash_clmulni_intel mpt3sas aesni_intel crypto_simd xhci_pci raid_class scsi_transport_sas cryptd xhci_hcd dm_mod
[ 3491.297280] ---[ end trace 449aff45c1ad5d1b ]---
[ 3491.297285] RIP: 0010:__btree_iter_traverse_all (fs/bcachefs/btree_iter.c:1332 (discriminator 1))
[ 3491.297291] Code: 00 00 00 0f 85 17 01 00 00 48 83 c4 78 44 89 f0 5b 5d 41 5c 41 5d 41 5e 41 5f c3 be 01 00 00 00 e8 67 fc ff ff e9 56 fe ff ff <0f> 0b 48 8b 73 08 65 8b 05 2d 92 bb 5c 89 c0 48 0f a3 05 ab 3d 0e
All code
========
   0:   00 00                   add    %al,(%rax)
   2:   00 0f                   add    %cl,(%rdi)
   4:   85 17                   test   %edx,(%rdi)
   6:   01 00                   add    %eax,(%rax)
   8:   00 48 83                add    %cl,-0x7d(%rax)
   b:   c4                      (bad)
   c:   78 44                   js     0x52
   e:   89 f0                   mov    %esi,%eax
  10:   5b                      pop    %rbx
  11:   5d                      pop    %rbp
  12:   41 5c                   pop    %r12
  14:   41 5d                   pop    %r13
  16:   41 5e                   pop    %r14
  18:   41 5f                   pop    %r15
  1a:   c3                      retq
  1b:   be 01 00 00 00          mov    $0x1,%esi
  20:   e8 67 fc ff ff          callq  0xfffffffffffffc8c
  25:   e9 56 fe ff ff          jmpq   0xfffffffffffffe80
  2a:*  0f 0b                   ud2             <-- trapping instruction
  2c:   48 8b 73 08             mov    0x8(%rbx),%rsi
  30:   65 8b 05 2d 92 bb 5c    mov    %gs:0x5cbb922d(%rip),%eax        # 0x5cbb9264
  37:   89 c0                   mov    %eax,%eax
  39:   48                      rex.W
  3a:   0f                      .byte 0xf
  3b:   a3                      .byte 0xa3
  3c:   05                      .byte 0x5
  3d:   ab                      stos   %eax,%es:(%rdi)
  3e:   3d                      .byte 0x3d
  3f:   0e                      (bad)

Code starting with the faulting instruction
===========================================
   0:   0f 0b                   ud2
   2:   48 8b 73 08             mov    0x8(%rbx),%rsi
   6:   65 8b 05 2d 92 bb 5c    mov    %gs:0x5cbb922d(%rip),%eax        # 0x5cbb923a
   d:   89 c0                   mov    %eax,%eax
   f:   48                      rex.W
  10:   0f                      .byte 0xf
  11:   a3                      .byte 0xa3
  12:   05                      .byte 0x5
  13:   ab                      stos   %eax,%es:(%rdi)
  14:   3d                      .byte 0x3d
  15:   0e                      (bad)
[ 3491.297301] RSP: 0018:ffffba80507e7838 EFLAGS: 00010202
[ 3491.297306] RAX: ffff939c650a4000 RBX: ffffba80507e7a38 RCX: 0000000000000001
[ 3491.297311] RDX: 0000000000000002 RSI: 0000000000000005 RDI: ffff939c650a4000
[ 3491.297317] RBP: ffffba80507e7848 R08: 0000000000000000 R09: 0000000000000000
[ 3491.297322] R10: 0000000000000001 R11: 0000000000000000 R12: 0000000000000001
[ 3491.297327] R13: ffffffffffffffff R14: 0000000000000000 R15: ffff939c650a41d0
[ 3491.297333] FS:  00007f4404ed3b80(0000) GS:ffff939ce4a00000(0000) knlGS:0000000000000000
[ 3491.297340] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 3491.297347] CR2: 00000000d6f1b000 CR3: 00000001a9be0001 CR4: 00000000003706e0
[ 3491.297353] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 3491.297358] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[ 3496.068532] bcachefs (f6774f6f-4ce0-483a-b58c-bd43e48f9d1d): copygc finished but 90925/90925 sectors, 22784/22784 buckets not moved (move stats: moved 0 sectors, raced 0 keys, 0 sectors)
[ 3551.687472] bcachefs (f6774f6f-4ce0-483a-b58c-bd43e48f9d1d): copygc finished but 90925/90925 sectors, 22784/22784 buckets not moved (move stats: moved 0 sectors, raced 0 keys, 0 sectors)
[ 3565.223962] bcachefs (dm-3): copygc finished but 381808/381808 sectors, 49152/49152 buckets not moved (move stats: moved 0 sectors, raced 0 keys, 0 sectors)
```

**Comment by jpsollie** [comment]
quite sure you're hitting #294, watch https://github.com/koverstreet/bcachefs/issues/294#issuecomment-894366026

**Comment by koverstreet** [comment]
Fixed in fd3fdd74af - it was an incorrect assertion
